Javascript/shop.js

```diff
@@ -9,127 +9,213 @@ const instantchallengedesc = "Instead of needing enough coins to get autocomplet
+          player.worlds -= 35;
+          player.shopUpgrades.offeringPotion += 1;
+        }
+        break;
+      case 2:
+        if (player.worlds >= shopBaseCosts.obtainiumPotion) {
+          player.worlds -= 35;
+          player.shopUpgrades.obtainiumPotion += 1;
+        }
+        break;
+      case 3:
+        if (player.worlds >= (shopBaseCosts.offerTimer + 25 * player.shopUpgrades.offeringTimerLevel) && player.shopUpgrades.offeringTimerLevel < 7) {
+          player.worlds -= (shopBaseCosts.offerTimer + 25 * player.shopUpgrades.offeringTimerLevel);
+          player.shopUpgrades.offeringTimerLevel += 1;
+        }
+        break;
+      case 4:
+        if (player.worlds >= (shopBaseCosts.offerAuto + 25 * player.shopUpgrades.offeringAutoLevel) && player.shopUpgrades.offeringAutoLevel < 7) {
+          player.worlds -= (shopBaseCosts.offerAuto + 25 * player.shopUpgrades.offeringAutoLevel);
+          player.shopUpgrades.offeringAutoLevel += 1;
+        }
+        break;
+      case 5:
+        if (player.worlds >= (shopBaseCosts.obtainiumTimer + 25 * player.shopUpgrades.obtainiumTimerLevel) && player.shopUpgrades.obtainiumTimerLevel < 7) {
+          player.worlds -= (shopBaseCosts.obtainiumTimer + 25 * player.shopUpgrades.obtainiumTimerLevel);
+          player.shopUpgrades.obtainiumTimerLevel += 1;
+        }
+        break;
+      case 6:
+        if (player.worlds >= (shopBaseCosts.obtainiumAuto + 25 * player.shopUpgrades.obtainiumAutoLevel) && player.shopUpgrades.obtainiumAutoLevel < 7) {
+          player.worlds -= (shopBaseCosts.obtainiumAuto + 25 * player.shopUpgrades.obtainiumAutoLevel);
+          player.shopUpgrades.obtainiumAutoLevel += 1;
+        }
+        break;
 
+      case 7:
+        if (player.worlds >= shopBaseCosts.instantChallenge && !player.shopUpgrades.instantChallengeBought) {
+          player.worlds -= 300;
+          player.shopUpgrades.instantChallengeBought = true;
+        }
+        break;
+      case 8:
+        if (player.worlds >= (shopBaseCosts.cashGrab + 100 * player.shopUpgrades.cashGrabLevel) && player.shopUpgrades.cashGrabLevel < 7) {
+          player.worlds -= (shopBaseCosts.cashGrab + 100 * player.shopUpgrades.cashGrabLevel);
+          player.shopUpgrades.cashGrabLevel += 1;
         }
-        revealStuff();
+        break;
+      case 9:
+        if (player.worlds >= (shopBaseCosts.antSpeed + 200 * player.shopUpgrades.antSpeedLevel) && player.shopUpgrades.antSpeedLevel < 3) {
+          player.worlds -= (shopBaseCosts.antSpeed + 200 * player.shopUpgrades.antSpeedLevel);
+          player.shopUpgrades.antSpeedLevel += 1;
+        }
+        break;
+      case 10:
+        if (player.worlds >= 1500 && !player.shopUpgrades.talismanBought) {
+          player.worlds -= 1500;
+          player.shopUpgrades.talismanBought = true;
+        }
+        break;
+
     }
+    revealStuff();
+  }
 }
 
 function useConsumable(i) {
-    let p = true

```

Comment:
Add a line here at the top for to check if the user has no potions -- then it will not run this function

Reply:
Done, moved the potion check from the switch case to the top.